src/Plugins/PluginConfig.cs

```diff
@@ -61,6 +62,13 @@ public PluginConfig(string mainAssemblyPath)
         /// </summary>
         public bool PreferSharedTypes { get; set; }
 
+        /// <summary>
+        /// If set, replaces the default <see cref="AssemblyLoadContext"/> used by the <see cref="PluginLoader"/>.
+        /// Use this feature if the <see cref="AssemblyLoadContext"/> of the <see cref="Assembly"/> is not the Runtime's default load context.
+        /// i.e. (AssemblyLoadContext.GetLoadContext(Assembly.GetExecutingAssembly) != <see cref="AssemblyLoadContext.Default"/>
+        /// </summary>
+        public AssemblyLoadContext DefaultContext { get; set; } = AssemblyLoadContext.Default;
```

Comment:
What do you think about making the default value `AssemblyLoadContext.GetLoadContext(Assembly.GetExecutingAssembly)`? I'd like to see if that "just works" in most normal use cases. I have a hunch it would, but we should test that.

Reply:
I think that is a great idea that I didn't think of surprisingly, will very likely not break anything either.

When I was making this PR, I was only thinking from the perspective of "Add Extra Feature", which I guess made me a bit narrow minded; I think this is a better default.

It saves the user from having to manually set the context to this context in the configuration.